DataDog/datadog-ci-rb
# do not instantiate TestVisibility::Recorder unless CI visibility is enabled

**What does this PR do?**
Adds a NullRecorder subclass of TestVisibility::Recorder with an empty constructor and with all methods skipping tracing. When CI mode is not enabled, NullRecorder is being created to make Datadog::CI public interface work without errors.

**Motivation**
Recent case of TestVisibility::Recorder emitting logs in production when validating environment variables

**How to test the change?**
Unit tests are provided. Run a rails app with tracing enabled and check that there are no logs coming from CI module.
